challenges/process-multivalue-reference/README.md

```diff
@@ -0,0 +1,20 @@
+---
+tags:
+  - iteration
+---
+
+# Process Multivalue References
+
+In hierarchical documents, it is currently not completely clear how to process multivalue references.
+
+## Extensions
+
+- What if you want to parse a data element and this results in multiple values? see `input-1, tags`?
+- What if you want to generate an RDF:List from a multiple result? see `output, ex:hasAuthors`.
+- What if you want create additional resources based on a (processed) reference that restuls in multiple values?
+
+## Discussions
+
+- This is very related to 'multivalue-references', e.g., being able to create an rdf:List: is this part of this challenge, or part of 'multivalue-references'?
```

Comment:
I'm not completely sure that this is an independent challenge as you said.
If the challenge is to generate RDF containers or collections, I think this should be part of https://github.com/kg-construct/mapping-challenges/tree/main/challenges/rdf-collections.
Another challenge could be the way to parse data structures such as "input-1, tags" where there are multi-values in the same key/column.
IMO they are two different things, one is the output of RDF collections (that can be created from different data structures), and the other is the parsing of these structures. But we should understand if this is a pre-processing and preparation step or should be part of the mapping parsers

Reply:
I think generating an RDF list out of multiple returned values is one case but one might as well just want to handle the multiple values as alternative RDF terms. If we set this challenge under the RDF list generation from multiple values that would mean that we force the generation of lists/collections when we have multiple values, but I think it is better to give the option of choosing

Reply:
> we should understand if this is a pre-processing and preparation step or should be part of the mapping parsers

I think how/where something is executed should be independent of its definition. This is also what FunMap taught us. The same mapping rules can be processed either inline as it traditionally happened or as a preprocessing step. The rules definition allows us to chose what is the best for each case

Reply:
Totally agree that representation should be independent of the execution. 
My reflection is: if we are going to include all the necessary pre-processing and cleaning declarations in the same spec as the transformations rules, the documents are going to be huge, hard to create and maintain, etc.  It would be like to have CSVW and RML together. But that is another discussion that shouldn't be here ;-)

Reply:
but multiple values returned is not a cleansing task.

Reply:
in morph-csv (over tabular data), we treat the problem such as `"tags": "tag1, tag2, tag3"` as a preparation step (normalization to be more specific). The info about the separator came from CSVW annotations.

IMO, although it is a valid data structure it shouldn't be considered correct as input for mapping rules. If we follow database good practices, multi-value fields should be normalized before doing any "query". Another possibility would be to consider multi-value fields as an inherent feature of the input files, so mapping rules and parsers should consider it.

Reply:
it sounds good what you do with morph-csv, but that's not applicable in the case of streaming data because there you can't have such preprocessing step.  

I think it's important to distinguish the declaration from the execution. We may declare where a certain processing needs to happen but we may _chose_ where to execute such processing, either as preprocessing step or as an "in-line" task. 

I think the aforementioned fits well with the discussions we had for sustainable workflows, ie either it's a preprocessing step or an in-line step, it needs to be declaratively defined to be reproducible.